src/main/java/org/osiam/client/OsiamConnector.java

```diff
@@ -170,31 +173,52 @@ private OsiamConnector(Builder builder) {
         authServiceEndpoint = builder.authServiceEndpoint;
         resourceServiceEndpoint = builder.resourceServiceEndpoint;
         genEndpoint = builder.genEndpoint;
-    }
 
-    /**
-     *
-     * @return a valid AuthService build out of the provided variables
-     */
-    private AuthService authService() {// NOSONAR - its ok if the Cyclomatic Complexity is > 10
-        if (authService == null) {
-            AuthService.Builder builder = new AuthService.Builder(getAuthServiceEndpoint());
+        String authServiceEndpoint = getAuthServiceEndpoint();
 
+        if (!Strings.isNullOrEmpty(authServiceEndpoint)) {
+            AuthService.Builder authServiceBuilder = new AuthService.Builder(authServiceEndpoint);
             if (clientId != null) {
-                builder = builder.setClientId(clientId);
+                authServiceBuilder = authServiceBuilder.setClientId(clientId);
             }
             if (clientSecret != null) {
-                builder = builder.setClientSecret(clientSecret);
+                authServiceBuilder = authServiceBuilder.setClientSecret(clientSecret);
             }
             if (clientRedirectUri != null) {
-                builder = builder.setClientRedirectUri(clientRedirectUri);
+                authServiceBuilder = authServiceBuilder.setClientRedirectUri(clientRedirectUri);
             }
+            authService = authServiceBuilder.build();
+        }
+
+        String resourceServiceEndpoint = getResourceServiceEndpoint();
+
+        if (!Strings.isNullOrEmpty(resourceServiceEndpoint)) {
+            userService = new OsiamUserService.Builder(resourceServiceEndpoint).build();
+            groupService = new OsiamGroupService.Builder(resourceServiceEndpoint).build();
+        }
+    }
 
-            authService = builder.build();
+    private AuthService authService() {
+        if (null == authService) {

```

Comment:
I'm no big fan of [Yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions) I think it reduces readability significantly and also we don't have it anywhere else in the code base. I would suggest that the condition is reversed:
```
if (authservice == null) {
```